ha_healthchecker/README.md

```diff
@@ -0,0 +1,170 @@
+
+    * `restarting`: It means the service hits error and `ha_healthchecker` will try to restart it to solve the problem.
+
+    * `down`: It means the service hits error. `ha_healthchecker` can't solve the problem by restarting the service. The process status in systemd is not `running`. It may be caused by many cases, operator should debug by hand.
+
+## How it works
+
+`ha_healthchecker` now includes **Refresh**, **Fix** and **Switch** functions. More will be added in the future if needed.
+
+* **Refresh**
+
+    `ha_healthchecker` checks OpenLab nodes and the services which run on them every 2 minutes. If everything is OK, the nodes/services' heartbeat will be refreshed. Otherwise the service will be marked as `restarting` or `down`.
+
+* **Fix**
+
+    Once any service hits error(that's said the service status in systemd is not **running**), `ha_healthchecker` will do the fix step. Here are some cases:
+
+    1. If the service is a necessary service.
+  
+        a. If the service is marked as `restaring`, it'll be restarted max 3 times (by default).
+
+        b. If the service is marked as `down`, `ha_healthchecker` will send a github issue to `openlab` repo and try to start switching Master and Slave.
+
+    2. If the service is an unnecessary service.
+
+        a. If the service is marked as `restarting`, it'll be restarted as well.
+
+        b. If the service is marked as `down` but the the time doesn't hit the switch deadline(48 hours by default), `ha_healthchecker` will only send out a github issue.
+
+        c. If the service is marked as `down` and the the time hits the switch deadline, `ha_healthchecker` will try to start switching Master and Slave.
+
+* **Switch**
+
+    Once **Fix** tell that OpenLab HA deployment need switch Master and Slave, and the configuration `allow_switch` is `True`. `ha_healthchecker` will send a github issue to `openlab` repo and do the switch work.
+
+## Configuration
+
+There are some configurations that used by `ha_healthchecker`. print `openlab ha config list` to get the value.
+
+| Name | Default Value | Description |
+| ---- | ------------- | ----------- |
+| allow_switch | False | Whether allow switch the HA deployment or not. |
+| dns_log_domain | test-logs.openlabtesting.org | The log service domain name. Usually it's `logs.openlabtesting.org`. |
+| dns_master_public_ip | None | The IP of the master node that runs zuul-web. |
+| dns_provider_account | None | The dns server account  for login. OpenLab use simpleDNS by default. |
+| dns_provider_api_url | https://api.dnsimple.com/v2/ | The dns server API url for login. |
+| dns_provider_token | None | The dns server token for login. |
+| dns_slave_public_ip | None | The IP of the slave node that runs zuul-web.  |
+| dns_status_domain | test-status.openlabtesting.org' | The zuul status web domain name. Usually it's `status.openlabtesting.org`. |
+| github_app_name | None | The github app that used by openlab CI, Usually it's `theopenlab-ci`. |
```

Comment:
why `theopenlab-ci` and `theopenlab/openlab` is not the default values.

Reply:
For testing case, If we forget to update config, it may impact openlab production environment. So I make many options be test value or None value